knowledge_repo/app/auth_providers/ldap.py

```diff
@@ -0,0 +1,25 @@
+from flask import request, render_template
+from ldap3 import Server, Connection, ALL
+
+from ..models import User
+from ..auth_provider import KnowledgeAuthProvider
+
+
+class LdapAuthProvider(KnowledgeAuthProvider):
+    _registry_keys = ['ldap']
+
+    def init(self):
+        self.server = Server(self.app.config['LDAP_SERVER'], get_info=ALL)
```

Comment:
Can we add a check here to see if LDAP_SERVER is in the config and not `None`, and if so, raise a polite error? [I'll be suggesting we don't set an LDAP_SERVER by default below].

Maybe:
```
if not self.app.config.get('LDAP_SERVER'):
    raise RuntimeErrror("Use of LDAP authentication requires specification of the LDAP_SERVER configuration variable.")
self.server = ...
```